extensions/interactions/FractionInput/FractionInputValidationService.js

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Validator service for the fraction interaction.
+ */
+
+oppia.factory('FractionInputValidationService', [
+  'WARNING_TYPES', 'baseInteractionValidationService', 'FractionObjectFactory',
+  function(
+    WARNING_TYPES, baseInteractionValidationService, FractionObjectFactory) {
+    return {
+      getCustomizationArgsWarnings: function(customizationArgs) {
+        return [];
+      },
+      getAllWarnings: function(
+        stateName, customizationArgs, answerGroups, defaultOutcome) {
+        var warningsList = [];
+
+        warningsList = warningsList.concat(
+          this.getCustomizationArgsWarnings(customizationArgs));
+
+        var toFloat = function(fraction) {
+          return FractionObjectFactory.fromDict(fraction).toFloat();
+        };
+        /*
```

Comment:
FYI, generally we either use // comments, or we do:

    /**
     * ...
     * ...
     * ...
     */

Former is slightly preferable, but either is fine.

Reply:
Done.